pkg/pillar/types/dns.go

```diff
@@ -732,7 +751,7 @@ func GetPortCost(dns DeviceNetworkStatus, ifname string) uint8 {
 
 func GetPort(dns DeviceNetworkStatus, ifname string) *NetworkPortStatus {
 	for _, us := range dns.Ports {
-		if us.IfName != ifname {
+		if ifname != "" && us.IfName != ifname {
```

Comment:
Do we have callers which pass an empty string as the ifname argument? If they do this will match on the first management port which might not be what we want.

Reply:
getLocalAddrListImpl() only calls it with a non empty ifname.

Could VerifyAllIntf() pass an empty string?